networkit/cpp/centrality/ApproxElectricalCloseness.cpp

```diff
@@ -0,0 +1,669 @@
+    tFinishGlobal.resize(threads, std::vector<count>(n));
+    generators.reserve(threads);
+    for (omp_index i = 0; i < threads; ++i)
+        generators.emplace_back(Aux::Random::getURNG());
+
+    degDist.reserve(G.upperNodeIdBound());
+    G.forNodes([&](node u) { degDist.emplace_back(0, G.degree(u) - 1); });
+
+    ustChildPtrGlobal.resize(threads, std::vector<node>(n));
+    ustSiblingPtrGlobal.resize(threads, std::vector<node>(n));
+    bfsParent.resize(n, none);
+}
+
+count ApproxElectricalCloseness::computeNumberOfUSTs() const {
+    return rootEcc * rootEcc
+           * static_cast<count>(
+               std::ceil(std::log(2.0 * static_cast<double>(G.numberOfEdges()) / delta)
+                         / (2.0 * epsilon * epsilon * (1. - kappa) * (1. - kappa))));
+}
+
+node ApproxElectricalCloseness::approxMinEccNode() {
+    auto &status = statusGlobal[0];
+    std::vector<count> distance(G.upperNodeIdBound());
+    std::vector<count> eccLowerBound(G.upperNodeIdBound());
+
+    auto maxDegreeNode = [&]() -> node {
+        node maxDegNode = 0;
+        count maxDeg = 0;
+        G.forNodes([&](const node u) {
+            const auto degU = G.degree(u);
+            if (degU > maxDeg) {
+                maxDeg = degU;
+                maxDegNode = u;
+            }
+        });
+        return maxDegNode;
+    };
+
+    auto doBFS = [&](const node source) -> node {
+        std::queue<node> q;
+        q.push(source);
+        status[source] = NodeStatus::VISITED;
+        distance[source] = 0;
+        node farthest = 0;
+
+        do {
+            const node u = q.front();
+            q.pop();
+            eccLowerBound[u] = std::max(eccLowerBound[u], distance[u]);
+            farthest = u;
+
+            G.forNeighborsOf(u, [&](const node v) {
+                if (status[v] == NodeStatus::NOT_VISITED) {
+                    q.push(v);
+                    status[v] = NodeStatus::VISITED;
+                    distance[v] = distance[u] + 1;
+                }
+            });
+
+        } while (!q.empty());
+
+        std::fill(status.begin(), status.end(), NodeStatus::NOT_VISITED);
+        return farthest;
+    };
+
+    node source = maxDegreeNode();
+
+    for (uint32_t i = 0; i < sweeps; ++i)
+        source = doBFS(source);
+
+    // Return node with minimum ecc lower bound
+    return static_cast<node>(std::min_element(eccLowerBound.begin(), eccLowerBound.end())
+                             - eccLowerBound.begin());
+}
+
+void ApproxElectricalCloseness::computeNodeSequence() {
+    // We use thread 0's vector
+    auto &status = statusGlobal[0];
+
+    // Compute the biconnected components
+    auto &bcc_ = *(bcc.get());
+    bcc_.run();
+    auto components = bcc_.getComponents();
+
+    std::queue<node> queue;
+    std::vector<node> curSequence;
+
+    for (const auto &curComponent : components) {
+        // Biconnected components with 3 nodes or less can be handled trivially.
+        if (curComponent.size() == 2) {
```

Comment:
In a previous implementation I also had a specialization for 3 vertices, which was later removed because it complicates the code without bringing better performance. I'll update the comment.